text/0039-service-overhaul.md

```diff
@@ -0,0 +1,462 @@
+    - `"keyint"` (optional): default keyframe interval for this service.
+    - `"bframes"` (optional): default b-frames for this service.
+    - `"profile"` (optional): default H264 encoder profile for this service.
+    - `"x264opts"` (optional): default option for x264 for this service.
+    - `"max video bitrate"` (optional): maximum video bitrate for this service.
+    - `"max audio bitrate"` (optional): maximum audio bitrate for this service.
+    - `"max fps"` (optional): maximum frame per second for this service.
+    - `"supported resolutions"` (optional): array of supported resolutions for this service.
+    - `"bitrate matrix"` (optional): array of maximum bitrate for a certain resolution/FPS combo for this service.
+      - `"res"`: resolution.
+      - `"fps"`: frame per seconds.
+      - `"max bitrate"`: maximum video bitrate.
+
+##### Issues with this format
+- The naming scheme is a mix of object name with space and some with underscores.
+- About `"output"`, OBS consider every service as RTMP if not added and it's not a recommended settings at all. It makes OBS use the right protocol for the service. Also this prevent a service of being multi-protocol.
+- About `"recommended"`, most of the options seems to H264 related
+- `"common"`, the name makes it not understandable at the first sight maybe adding some documention would be good thing.
+
+#### New format (WIP)
+The new format will be parsable with a new interface library named `service-json-parser` to allow first-party plugins to read this format rather than recreating the wheel.
+
+The new format will be considered as the version 4.
+
+This format allow some specific additions parsed by getter added to the concerned plugin.
+
+Multi-service plugins should have a JSON root that look like this:
+```json
+{
+    "format_version": 4,
+    "services": []
+}
+```
+`"services"` is an array of service objects.
+
+Service plugin like integrations one should have a JSON root that look like this:
+```json
+{
+    "format_version": 4,
+    "service": {}
+}
+```
+`"service"` is the service object itself.
+
+##### Plugins specific additions
+Those additions are added like extensions to the service object format.
+
+Getters for those customisations are implemented in the plugin itself.
+- In `obs-services`:
+  - `"id"` (required): service identifier, services are no longer identified by their name. Added because this plugin is multi-services.
+  - `"name"` (required): name of the service. Added because this plugin is multi-services.
+  - `"common"` (optional, false by default): this object is added in service objects of this plugin to identify which service in "common" or not. No third-party plugin should have this behavior.
+- In `obs-youtube`:
+  - Servers name are translated.
+  - `"name_suffix"` in `"servers"` (optional): suffix added to the server name after being translated.
```

Comment:
What is a use case for such a suffix?

Reply:
Those names are used for YouTube servers whatever the protocol:
- "Primary YouTube ingest server"
- "Backup YouTube ingest server"

And for RTMP they add (legacy RTMP) at the end ("Primary YouTube ingest server (legacy RTMP)").

I wanted to add translations for those two names. And put (legacy RTMP) as a suffix to have 2 translation keys rather than 4.

I will add the use case on the RFC.

Reply:
`"name_suffix"` in servers will may be removed and be replaced by a possibility to change the protocol name in the protocol drop-down (will not change the value). In YouTube case the drop-down will show "Legacy RTMP" rather than "RTMP".

Reply:
Completely removed, and use 4 translations key.